### DISCUSSION TITLE
docker-compose examples for reverse proxies and other guides

### DISCUSSION THREAD

**Discussion by ho16** [nextcloud/all-in-one#1475] [Questions] [answered]
Hello,

in the past I had a setup  with 3 docker containers letsencrypt on a nginx reverse proxy, mariadb and  nextcloud. 
All three of them were used from linuxserver.io. It worked for many years without bigger problems. 
(For sure it was a very small installation for up to 5 people.)
Now I wanted to go to the "nextcloud all in one solution§, all running in docker containers as  well.

Does anyone have a working solution with a reverse proxy, with integrated letsencrypt functionality? 
I used it with a dyndns solution and a sudomain configuration.
I am not fixed on the nginx solution, when there is another one working with the "all in one" image. 

What do you suggest, what is recomended for the installation?

During my tests with image linuxserver/swag, it seems to me that the recomended configurations did not work on it, 

Thanks

**Comment by szaimen** [comment, marked as answer]
> What do you suggest, what is recomended for the installation?

I would usually recommend caddy which is the easiest one. See e.g. https://github.com/nextcloud/all-in-one/discussions/575#discussion-4055615

**Reply by ho16** [reply]
> > What do you suggest, what is recomended for the installation?
> 
> I would usually recommend caddy which is the easiest one. See e.g. [#575 (comment)](https://github.com/nextcloud/all-in-one/discussions/575#discussion-4055615)

Ok, but as far as I read, there is no letsencrypt functionality in the caddy docker image included, is it? How do you handle that?

**Reply by szaimen** [reply]
Caddy handles that for you automatically

**Reply by ho16** [reply]
> Caddy handles that for you automatically

Ok, that sounds great, I will give it a try and show my results here. 

Thank you very much

**Reply by ho16** [reply]
> Caddy handles that for you automatically

Hello,

after some testings the docker image for caddy is working. I am coming to the point where a letsencrypt certficate is assigned and I get a response message from the Caddyfile in the browser. 
So for now the proxy is running in a docker container. 
Thanks for the tips.

Now I have other problems. (I will write it in a new question.)

Thanks a lot.

**Reply by vawaver** [reply]
I use Nextcloud AIO docker + Nginx Reverse Proxy server (docker as well).
Everything is working fine for me with the right settings.
Let me know if you solved the problem.
If you do not, I can share my settings.

**Reply by ho16** [reply]
It is fixed in the recomended setup. Caddy as reverse proxy and the normal setup,.
Thanks for the assistance.
